lib/tunestakeoutwrapper.rb

```diff
@@ -0,0 +1,53 @@
+# this is the one that is talking to Charles API
+require 'httparty'
+# module
+module TunesTakeout
+  BASE_URL = "https://tunes-takeout-api.herokuapp.com/"
+  attr_reader :search, :top_favorites_by, :final_results
+
+  def self.search(query, limit = 10, seed = query)
+    # /v1/suggestions/search?query=banana
+    # find a tune with the passed query
+    data = HTTParty.get(BASE_URL + "v1/suggestions/search?query=#{query}&limit=#{limit}&seed=#{query}").parsed_response
+  end
+
+  def self.top_favorites_by
+    # hard coded limit of 20, possible refractor would be to make this an argument.
+    data = HTTParty.get(BASE_URL + "v1/suggestions/top?limit=20").parsed_response
+  end
+
+  def self.final_results(suggestions_hash)
+    # sending an id and retrieving the pairings for search
+    @final_results = []
+    suggestions = suggestions_hash["suggestions"]
+    suggestions.each do |id|
+      data = HTTParty.get(BASE_URL + "v1/suggestions/#{id["id"]}").parsed_response
+      @final_results << data
+    end
+    @final_results
+  end
+
+  def self.final_results_with_id(results)
+    # sending an id and retrieving the pairings for top 20
+    @top_results = []
```

Comment:
Thank you for looking through my code! I will make some edits.

Risha

On Tue, May 24, 2016 at 3:55 PM, Daphne <notifications@github.com> wrote:

> In lib/tunestakeoutwrapper.rb
> <https://github.com/Ada-C5/tunes-takeout/pull/19#discussion_r64489574>:
>
> > +  end
> > +
> > +  def self.final_results(suggestions_hash)
> > +    # sending an id and retrieving the pairings for search
> > +    @final_results = []
> > +    suggestions = suggestions_hash["suggestions"]
> > +    suggestions.each do |id|
> > +      data = HTTParty.get(BASE_URL + "v1/suggestions/#{id["id"]}").parsed_response
> > +      @final_results << data
> > +    end
> > +    @final_results
> > +  end
> > +
> > +  def self.final_results_with_id(results)
> > +    # sending an id and retrieving the pairings for top 20
> > +    @top_results = []
>
> Again here with the instance variables -- can be local variable instead.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Ada-C5/tunes-takeout/pull/19/files/52a7263730cb0a2078eaeb092ec6181158bafce7#r64489574>
>